### ISSUE TITLE
mobx-persist-store stable release

### ISSUE THREAD

**Issue by quarrant** [quarrant/mobx-persist-store#36]
Supports only mobx v6

**API**
- [ ] makeAutoPersist
- [ ] isSynchronized
- [ ] waitSynchronization
- [ ] forceSynchronize
- [ ] startPersist
- [ ] stopPersist
- [ ] clearStorage
- [ ] configure

**New**
- skip non observable properties for economy memory
- set global adapter for all stores (also can use local adapter)

**Comment by quarrant** [comment]
@codeBelt FYI

**Comment by codeBelt** [comment]
@quarrant is this a todo list for a v1.0.0 release?

**Comment by quarrant** [comment]
@codeBelt yep

**Comment by codeBelt** [comment]
@quarrant can you write more information on what these are:

- makeAutoPersist
- waitSynchronization
- forceSynchronize
- skip non observable properties for economy memory
- set global adapter for all stores (also can use local adapter)

**Comment by quarrant** [comment]
@codeBelt 
- makeAutoPersist (like makeAutoObservable =)) - configure persist store with params: name, properties (uses annotations), storage (without StorageAdapter class) and reactionOptions
- waitSynchronization - promise that waiting when all processes in makeAutoPersist will be done
- forceSynchronize - trigger to load storage into store (issue #23)
- skipNonObservable - if set `true` non observable properties doesn't save to storage (memory economy) 
- you can set storage options uses configure and it will be work for all stores, but also you can override them in makeAutoPersist

**Comment by codeBelt** [comment]
@quarrant Here is what I have so far. I had a little help from a friend on naming.

- `makeAutoPersistable` (todo by @quarrant)
- `makePersistable`

With Required Options:

```ts
import { makePersistable } from 'mobx-persist-store';

makePersistable(this, { name: 'myStore', properties: ['someProperty'], storage: window.localStorage });
```

With All Options:

```ts
import { configure } from 'mobx-persist-store';
import localForage from "localforage";
import ms from 'milliseconds';

makePersistable(
  this,
  {
    name: 'myStore',
    properties: ['someProperty'],
    storage: localForage,
    expireIn: ms.days(7),
    removeOnExpiration: false,
    stringify: false,
  },
  { delay: 200 }, // Optional
);
```
### API 

- `clearPersistedStore`
- `hydrateStore`
- `isHydrated`
- `isPersisting`
- `pausePersisting`
- `startPersisting`
- `stopPersisting`


### Configuration
```ts
import { configure } from 'mobx-persist-store';
import localForage from "localforage";
import ms from 'milliseconds';

configure({
  storage: localForage, // localForage, window.localStorage, AsyncStorage all have the same interface
  expireIn: ms.days(7),
  removeOnExpiration: false,
  stringify: false,
  delay: 200,
});

...

// Only `name` and `properties` are needed for `makePersistable`
makePersistable(this, { name: 'myStore', properties: ['someProperty'] });
```

I would like to write tests for `PersistStore.ts` still before I hand it off to you. I hope to get that done next weekend.

**Comment by quarrant** [comment]
@codeBelt can we talk in telegram? 

https://t.me/quarrant

**Comment by codeBelt** [comment]
@quarrant I sent you a LinkIn request

**Comment by codeBelt** [comment]
@quarrant I think I am finished with all my code changes and update the README. https://github.com/codeBelt/mobx-persist-store/tree/MobxStorePersist#readme

The only thing I will need to do is update the example demo.

**Comment by codeBelt** [comment]
@quarrant Do want to merge my code and get a Release Candidate `1.0.0-rc.0` tag out there? Then I can start woking on the example demo and have people start playing with it.

- https://semver.npmjs.com/

**Comment by quarrant** [comment]
> @quarrant Do want to merge my code and get a Release Candidate `1.0.0-rc.0` tag out there? Then I can start woking on the example demo and have people start playing with it.
> 
> * https://semver.npmjs.com/

done

**Comment by codeBelt** [comment]
@quarrant what do you think of new code: https://codesandbox.io/s/mobx-persist-store-version-1-f852w?file=/src/stores/User.store.ts&theme=dark

Do you have anyone else to give us feedback on the new changes?

**Comment by codeBelt** [comment]
@quarrant Have you tried out `1.0.0-rc.0`? Is it ready to be released as `1.0`?

**Comment by quarrant** [comment]
@codeBelt yep! It's a great work! let's go =)
